File: 121_Best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;
        int buyPrice = prices[0];
        for (int i=1;i<prices.size();i++){
            if (prices[i]>buyPrice){
                maxProfit=max(maxProfit,prices[i]-buyPrice);
            }
            buyPrice= min(buyPrice,prices[i]);
        } 
        return maxProfit;
    }
};

File: Arrays/121BestTimetoBuyandSellStock.ts
function maxProfit(prices: number[]): number {
    let min=prices[0];//7
    let profit =0//0
    for(let i=1 ;i<prices.length;i++){
        let currprofit=prices[i]-min
        profit = Math.max(profit,currprofit)
        min = Math.min(min,prices[i])
    }
    return profit
};